### ISSUE TITLE
`no-extra-parens`: Conditional Spread Syntax

### ISSUE THREAD

**Issue by erosman** [eslint-stylistic/eslint-stylistic#664]
### Clear and concise description of the problem

Conditional spread syntax uses parentheses and should be allowed under the rule

Examples from [Spread syntax (...)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax)

```js
const isSummer = false;
const fruits = ["apple", "banana", ...(isSummer ? ["watermelon"] : [])];
// ['apple', 'banana']
```

```js
const isSummer = false;
const fruits = {
  apple: 10,
  banana: 5,
  ...(isSummer ? { watermelon: 30 } : {}),
};
```

```js
const isSummer = false;
const fruits = {
  apple: 10,
  banana: 5,
  ...(isSummer && { watermelon: 30 }),
};
```

[Playground](https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgaXNTdW1tZXIgPSBmYWxzZTtcbmNvbnN0IGZydWl0cyA9IHtcbiAgYXBwbGU6IDEwLFxuICBiYW5hbmE6IDUsXG4gIC4uLihpc1N1bW1lciA/IHsgd2F0ZXJtZWxvbjogMzAgfSA6IHt9KSxcbn07Iiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby1leHRyYS1wYXJlbnMiOlsiZXJyb3IiXX0sImxhbmd1YWdlT3B0aW9ucyI6eyJzb3VyY2VUeXBlIjoibW9kdWxlIiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==)

### Suggested solution

Either of:

- Update `no-extra-parens`
- Add a new option

### Additional context

See also: https://github.com/eslint-stylistic/eslint-stylistic/issues/310#issuecomment-2565490889

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue
